src/taipy/core/data/parquet.py

```diff
@@ -0,0 +1,166 @@
+        if properties is None:
+            properties = {}
+        if missing := set(self._REQUIRED_PROPERTIES) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties " f"{', '.join(x for x in missing)} were not informed and are required"
+            )
+
+        if self.__COLUMNS_PROPERTY not in properties.keys():
+            properties[self.__COLUMNS_PROPERTY] = None
+
+        if self.__COMPRESSION_PROPERTY not in properties.keys():
+            ...
+
+        self._path = properties.get(self.__PATH_KEY, properties.get(self.__DEFAULT_PATH_KEY))
+        if self._path is None:
+            raise MissingRequiredProperty("default_path is required in a Parquet data node config")
+        else:
+            properties[self.__PATH_KEY] = self._path
+
+        if self.__EXPOSED_TYPE_PROPERTY not in properties.keys():
+            properties[self.__EXPOSED_TYPE_PROPERTY] = self.__EXPOSED_TYPE_PANDAS
+        self._check_exposed_type(properties[self.__EXPOSED_TYPE_PROPERTY])
+
+        super().__init__(
+            config_id,
+            scope,
+            id,
+            name,
+            owner_id,
+            parent_ids,
+            last_edit_date,
+            job_ids,
+            cacheable,
+            validity_period,
+            edit_in_progress,
+            **properties,
+        )
+        if not self._last_edit_date and isfile(self._path):
+            self.unlock_edit()
+
+    @classmethod
+    def storage_type(cls) -> str:
+        return cls.__STORAGE_TYPE
+
+    @property  # type: ignore
+    @_self_reload(DataNode._MANAGER_NAME)
+    def path(self):
+        return self._path
+
+    @path.setter
+    def path(self, value):
+        self._path = value
+        self.properties[self.__PATH_KEY] = value
+
+    def _check_exposed_type(self, exposed_type):
+        if exposed_type not in self.__VALID_STRING_EXPOSED_TYPES:
+            raise InvalidExposedType(
+                f"Invalid exposed type {exposed_type}. Supported values are {', '.join(self.__VALID_STRING_EXPOSED_TYPES)}"
+            )
+
+    def _read(self):
+        if self.properties[self.__EXPOSED_TYPE_PROPERTY] == self.__EXPOSED_TYPE_PANDAS:
+            return self._read_as_pandas_dataframe()
+        if self.properties[self.__EXPOSED_TYPE_PROPERTY] == self.__EXPOSED_TYPE_MODIN:
+            return self._read_as_modin_dataframe()
+        if self.properties[self.__EXPOSED_TYPE_PROPERTY] == self.__EXPOSED_TYPE_NUMPY:
+            return self._read_as_numpy()
+
+    def _read_as_numpy(self):
+        return self._read_as_pandas_dataframe().to_numpy()
+
+    def _read_as_pandas_dataframe(self) -> pd.DataFrame:
+        return pd.read_parquet(self._path, columns=self.properties[self.__COLUMNS_PROPERTY])
+
+    def _read_as_modin_dataframe(self) -> modin_pd.DataFrame:
+        return modin_pd.read_parquet(self._path, columns=self.properties[self.__COLUMNS_PROPERTY])
+
+    def _write(self, data: Any):
```

Comment:
Can you add a `write_with_column_names()` like [this method in CSVDataNode](https://github.com/Avaiga/taipy-core/blob/develop/src/taipy/core/data/csv.py#L203), so that the user can write with different column name?

Reply:
Sure.

https://github.com/Avaiga/taipy-core/blob/7381d79e0dbd5324aaefb4b5f2f4bc3666bd8054/src/taipy/core/data/csv.py#L216-L218

I'm not sure what these lines are for though -- do I copy this as well?

Reply:
Line 216 is to update the last edit time made by Taipy.
Line 217-218 is to make sure that the list of jobs which are working on this data node is updated.

And yes, these line should be on ParquetDataNode too. Thank you 😁

Reply:
I think this functionality is quite specific and easily achievable in the Task // function rather than having a separate method for it. I've expressed this to @jrobinAV and I think he agrees, but do let me know if I should add it anyway.